docs/v1/api.md

```diff
@@ -0,0 +1,239 @@
+
+## Persistent Volumes
+
+First, this problem space is similar to that of persistent volumes. We model the data structures herein after those in persistent volumes primarily because we believe that consistency is a virtue. More importantly, the community benefits when similar problems have similar solutions.
+
+Specifically, we believe that the following existing types provide a good model for the mechanics of how the services offered by a broker can be used in Kubernetes:
+
+- `StorageClass`
+- `PersistentVolume`
+- `PersistentVolumeClaim`
+
+## Ingress Controllers
+
+Today, operators choose their own implementation of an ingress controller. Kubernetes gives the operator the ability to specify any ingress controller, and the complete freedom to implement it themselves.
+
+Similarly, permitting operators to implement or select their own service catalog controller would encourage freedom of choice and diversity. The analog to an ingress controller for the service catalog system is called a _service catalog controller_. Like ingress, these controllers run inside a standard pod. Anyone can write and deploy a service catalog controller, with no changes necessary to the Kubernetes install.
+
+Service catalog controllers vary, however, from ingress controllers, because a cluster can consist of many different controllers, each of which handles actions on a subset of the available service-related resources. For example, a cluster can run a service controller to handle AWS RDS databases, and a completely separate one to handle AWS Elastic Load Balancer instances.
+
+As we'll detail below, each service controller implementation must to conform to an implementation spec, but that spec allows for a heterogenous set of controllers in a cluster. We believe this allowance encourages flexibility in a cluster's service catalog.
+
+# Kubernetes Resources
+
+Given these inspirations, the following resources will be added to the Kubernetes API:
+
+- `ServiceClass`
+  - Describes a _kind_ of service. This resource is semantically similar to how a `StorageClass` defines a _kind_ of persistent volume, but we expect that, in practice, the kinds of services represented here will be more coarse grained than the kinds of volumes represented in `StorageClass`es
+  - References a service ID and plan ID, which is managed and understood by a service controller (see below)
+  - Created by an operator or tool
+- `ServiceInstance`
+  - An instance of a `ServiceClass` -- an in-cluster representation of a provisioned service
+  - Can be statically provisioned by an operator, to enable multiple consumers to bind to it (enables multi-tenancy)
+  - Can be dynamically provisioned and bound to by a consumer, to guarantee an app’s exclusive use of the resource (guarantees single-tenancy)
+- `ServiceInstanceClaim`
+  - Created by a consumer to bind to an existing ServiceInstance by name. As written above, this workflow enables multi-tenant usage of a resource
```

Comment:
I definitely understand the desire to follow Kubernetes conventions and I will yield to that if this is considered something fundamental to the kubernetes way of doing things.  Otherwise, we should probably do what is most simple and intuitive.  @bgrant0607, do you have any thoughts on this comment thread and the importance of the claim object in the act of provisioning and binding?